### ISSUE TITLE
Nullable value reported as "is not null" when it can be (and actually is)

### ISSUE THREAD

**Issue by odalet** [dotnet/roslyn#43052]
**Version Used**: 
VS 2019 16.4.3
**Steps to Reproduce**:

1. Create a new solution containing: 
* a class library (in which 'nullable' is not enabled)
* an application referencing the library (in which 'nullable' is enabled)
2. The class lib exposes a class with a rw `string` property that is not initialized:

```csharp
namespace Lib {
public class TestClass {
	public string CanBeNull { get; set; }
} }
```
3. The application instanciates the class and affects the string property to a variable

```csharp
static class Program {
	static void Main() {
		var c = new TestClass();
		_ = c.CanBeNull; // Reports "'CanBeNull' is not null here" instead of "can be null here"
	} }
```
**Expected Behavior**:

VS tooltip should report **can be null here** when hovering `c.CanBeNull`

**Actual Behavior**:

![image](https://user-images.githubusercontent.com/79333/78372595-c27bf000-75c9-11ea-9155-3f02a10b006e.png)

VS tooltip actually says **is not null here**

----

I initially posted this behavior in #41110, but this seems to be unrelated. So here is its dedicated issue. Over there (https://github.com/dotnet/roslyn/issues/41110#issuecomment-607950222) @TessenR commented that this behavior may be 'by design'... Honestly I don't see how this can be left as is: it is very misleading as developers if only and blindly relying on VS tooltips/warning won't ever spot there is something wrong in situations like the one I exhibit above.

Attached is the simple repro solution: 
[IssueRepro43052.zip](https://github.com/dotnet/roslyn/files/4427983/IssueRepro43052.zip)

**Comment by sharwell** [comment]
Duplicate of #36983
